app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -189,185 +190,263 @@ public function getCartDataJs($checkoutType = 'multi-page')
-                $onPaymentSubmit = '';
-                $success = '';
-                $close = '';
-            } else {
-                $check = Mage::getStoreConfig('payment/boltpay/check');
-                $onCheckoutStart = Mage::getStoreConfig('payment/boltpay/on_checkout_start');
-                $onShippingDetailsComplete = Mage::getStoreConfig('payment/boltpay/on_shipping_details_complete');
-                $onShippingOptionsComplete = Mage::getStoreConfig('payment/boltpay/on_shipping_options_complete');
-                $onPaymentSubmit = Mage::getStoreConfig('payment/boltpay/on_payment_submit');
-                $success = Mage::getStoreConfig('payment/boltpay/success');
-                $close = Mage::getStoreConfig('payment/boltpay/close');
-            }
+    /**
+     * Generate cart data
+     *
+     * @param $orderCreationResponse
+     * @return array
+     */
+    public function buildCartData($orderCreationResponse)
+    {
+        $authCapture = (Mage::getStoreConfigFlag('payment/boltpay/auto_capture') === self::AUTO_CAPTURE_ENABLED);
+
+        //////////////////////////////////////////////////////////////////////////
+        // Generate JSON cart and hints objects for the javascript returned below.
+        //////////////////////////////////////////////////////////////////////////
+        $cartData = array(
+            'authcapture' => $authCapture,
+            'orderToken' => ($orderCreationResponse) ? $orderCreationResponse->token: '',
+        );
+
+        // If there was an unexpected API error, then it was stored in the registry
+        if (Mage::registry("api_error")) {
+            $cartData['error'] = Mage::registry("api_error");
+        }
 
-            //////////////////////////////////////////////////////
-            // Generate and return BoltCheckout javascript.
-            //////////////////////////////////////////////////////
-            $immutableQuoteId = ($immutableQuote) ? $immutableQuote->getId() : -1;
+        return $cartData;
+    }
 
-            $checkForAdmin = '';
-            $onSuccessCallback = '';
-            $onClose = '';
-            if ($checkoutType === 'admin') {
-                $checkForAdmin = "if ((typeof editForm !== 'undefined') && (typeof editForm.validate === 'function')) {
-                    var bolt_hidden = document.getElementById('boltpay_payment_button');
-                    bolt_hidden.classList.remove('required-entry');
+    /**
+     * Generate BoltCheckout Javascript for output.
+     *
+     * @param $checkoutType
+     * @param $immutableQuote
+     * @param $hintData
+     * @param $cartData
+     * @return string
+     */
+    public function buildBoltCheckoutJavascript($checkoutType, $immutableQuote, $hintData, $cartData)
+    {
+        /* @var Bolt_Boltpay_Helper_Api $boltHelper */
+        $boltHelper = Mage::helper('boltpay');
+
+        $jsonCart = json_encode($cartData);
+        $jsonHints = '{}';
+        if (sizeof($hintData) != 0) {
```

Comment:
comment line below suggests we don't need this if statement - if $hintData is empty $jsonHints will be an empty object.